utils/blackbox_test.py

```diff
@@ -0,0 +1,96 @@
+from itertools import product
+from pathlib import Path
+import pytest
+
+import pyteal as pt
+
+from utils.blackbox import blackbox_pyteal
+
+from tests.compile_asserts import assert_teal_as_expected
+
+
+@pt.Subroutine(pt.TealType.none, input_types=[])
+def utest_noop():
+    return pt.Pop(pt.Int(0))
+
+
+@pt.Subroutine(
+    pt.TealType.none,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_noop_args(x, y, z):
+    return pt.Pop(pt.Int(0))
+
+
+@pt.Subroutine(pt.TealType.uint64, input_types=[])
+def utest_int():
+    return pt.Int(0)
+
+
+@pt.Subroutine(
+    pt.TealType.uint64,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_int_args(x, y, z):
+    return pt.Int(0)
+
+
+@pt.Subroutine(pt.TealType.bytes, input_types=[])
+def utest_bytes():
+    return pt.Bytes("")
+
+
+@pt.Subroutine(
+    pt.TealType.bytes,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_bytes_args(x, y, z):
+    return pt.Bytes("")
+
+
+@pt.Subroutine(pt.TealType.anytype, input_types=[])
+def utest_any():
+    x = pt.ScratchVar(pt.TealType.anytype)
+    return pt.Seq(x.store(pt.Int(0)), x.load())
+
+
+@pt.Subroutine(
+    pt.TealType.anytype,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_any_args(x, y, z):
+    x = pt.ScratchVar(pt.TealType.anytype)
+    return pt.Seq(x.store(pt.Int(0)), x.load())
+
+
+UNITS = [
+    utest_noop,
+    utest_noop_args,
+    utest_int,
+    utest_int_args,
+    utest_bytes,
+    utest_bytes_args,
+    utest_any,
+    utest_any_args,
+]
+
+
+@pytest.mark.parametrize("subr, mode", product(UNITS, pt.Mode))
+def test_blackbox_pyteal(subr, mode):
+    """
+    TODO: here's an example of issue #199 at play - (the thread-safety aspect):
+    compare the following!
+    % pytest -n 2 tests/integration/graviton_test.py::test_blackbox_pyteal
+    vs
+    % pytest -n 1 tests/integration/graviton_test.py::test_blackbox_pyteal
+    """
+    path = Path.cwd() / "tests" / "teal" / "blackbox" / "unit"
```

Comment:
@tzaffi What do you think of the following thought?

* Looking at the dir structure `tests/teal`, I feel it's difficult to discern where these files are used.
* Proposal:  Move `tests/teal/blackbox/unit` to `utils/resources` or a similarly named dir under `utils`.

Reply:
👍

Reply:
@tzaffi Thanks for the move in https://github.com/algorand/pyteal/pull/249/commits/5520d710950f5281e8ed2a9312c366c809001310.  Resolving.